src/hooks/users/useCurrentUser.ts

```diff
@@ -0,0 +1,8 @@
+import { Route as AuthRoute } from "@/routes/_auth/route";
+
+type CurrentUser = ReturnType<typeof AuthRoute.useRouteContext>["user"];
+
+export function useCurrentUser(): CurrentUser {
```

Comment:
For whatever reason, TypeScript is inferring `CurrentUser` as `any` (even though the `AuthRoute.useRouteContext` resolve to a very simple object, the underlying TypeScript annotation is very very complex, with multiple layers of generics. When a type is that complicated, TypeScript often gives up (often for performance reasons) when you call `typeof` on it).

So let's just actually explicitly add the return type as `User | undefined` rather than `CurrentUser`